### PR TITLE
add more attributes to default mapping; use mapping user provides if …

### PR DESCRIPTION
**Pull request by andyh565**
…overlaps with defaults

### CODE REVIEW

File: test/integration/fixtures/notify/firehose-create-lead.json [hunk 1]
```diff
@@ -35,10 +27,18 @@
     "operation": "setIfNull",
     "value": 1548709440
   },
```
Comment (andyh565) [inline test/integration/fixtures/notify/firehose-create-lead.json:29]: @SMK1085 for leads, we use this "setIfNull" operation. We don't use on users though. Do you know why? Will this be an issue for Bestmile?

**Comment by andyh565** [comment]
@SMK1085 @michaloo there a two issues that this fixes:

1. The utm fields are missing from the default mapping. 

2. The attribute mapper allows you specify the Hull field for default mappings, but then actually ignores the field for incoming users and uses the default. 
Ex) for Bestmile, they have 
{ name: "utm_campaign", hull: "traits_intercom/intercom_utm_campaign" } would not be mapped to "traits_intercom/intercom_utm_campaign", but to "traits_intercom/utm_campaign".

File: test/integration/fixtures/notify/firehose-create-lead.json (hunk 1 above)
Comment (michaloo) [reply test/integration/fixtures/notify/firehose-create-lead.json:29]: Hm, not sure what is the impact for Bestmile. But overall leads implementation was kind of experimental and limited. So we don't have separate attributes mapper for leads, that's why we just wanted to get some extra data for leads based on default mapping, but we didn't want that data to overwrite user data which takes precedence.

**Comment by michaloo** [comment]
I'm not sure if I understood 100%, but if I understood correctly then it makes sense. Default mapping is applied only if we don't have a given intercom field already mapped by the user in the settings.

**Comment by michaloo** [comment]
Yes, we have it on the roadmap. We wanted first to close the current AWS migration including moving Salesforce connector and then it will be one of the next steps to evaluate if it makes more sense to upgrade the intercom connector to the latest API version while moving to monorepo or if we should break it down further.

Given the current interest and state of the connector would you prioritize that work more than that?

**Comment by SMK1085** [comment]
> Yes, we have it on the roadmap. We wanted first to close the current AWS migration including moving Salesforce connector and then it will be one of the next steps to evaluate if it makes more sense to upgrade the intercom connector to the latest API version while moving to monorepo or if we should break it down further.

Makes sense, I think there are a few things which customers asked for including synchronization of accounts (without linking), so it might be smart to break it down in smaller releases.

> Given the current interest and state of the connector would you prioritize that work more than that?

We should probably re-evaluate the support tickets, customer discussions and onboardings which are related to Intercom to determine whether we need to prioritize it further and weigh it up against other initiatives. Without reviewing the big picture first, I think further prioritization would not be justified imo. WDYT @michaloo?

**Pull request merged**